playground/frontend/playground_components/lib/src/services/analytics/analytics_service.dart

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'dart:async';
+import 'dart:collection';
+
+import 'package:get_it/get_it.dart';
+import 'package:meta/meta.dart';
+
+import 'events/abstract.dart';
+
+/// Abstract class to submit analytics events.
+abstract class AnalyticsService {
+  AnalyticsEvent? _lastEvent;
+  AnalyticsEvent? get lastEvent => _lastEvent;
+
+  final _defaultEventParameters = <String, dynamic>{};
+
+  /// The parameters sent with all events.
+  ///
+  /// Individual event parameters have higher priority on collisions.
+  Map<String, dynamic> get defaultEventParameters =>
+      UnmodifiableMapView(_defaultEventParameters);
+
+  set defaultEventParameters(Map<String, dynamic> newValue) {
+    _defaultEventParameters.clear();
+    _defaultEventParameters.addAll(newValue);
+  }
+
+  static AnalyticsService get() {
+    return GetIt.instance.get<AnalyticsService>();
+  }
+
+  /// Sends [event] asynchronously without returning a [Future].
+  void sendUnawaited(AnalyticsEvent event) {
```

Comment:
Why can't we just have `Future<void> send(AnalyticsEvent event) async { ... }`?

Reply:
Because when we call it and discard the future, we get a `discarded_future` lint. We would have to do one of the following with it:
- Disable the lint project-wide, which we do not want to, because it guards from logic errors.
- Disable it at each occurrence with which is harder to read.